Algorithms/ValidParentheses.cs

```diff
@@ -22,22 +22,38 @@ public class ValidParentheses
 			charsOfClose.Add('[', ']');
 			int lengthString = stringCharacters.Length, i = 0;
 
-			while (i < lengthString)
-			{
-				if (stringCharacters[i] == '(' || stringCharacters[i] == '{' || stringCharacters[i] == '[')
+			while (i < lengthString) {
+				if (validOpenChars(stringCharacters, i))
 					characters.Push(stringCharacters[i]);
-				else if (characters.Count > 0)
-				{
+
+				else if (characters.Count > 0) {
 					if (charsOfClose[characters.Peek()] == stringCharacters[i])
 						characters.Pop();
 					else
 						return false;
-				}
-				else
+				} else
 					return false;
 				i++;
 			}
-			return characters.Count == 0 ? true : false;
+			return characters.Count == 0 & true;
+		}
+		/// <summary>
+		/// Valid open chars
+		/// </summary>
+		/// <param name="stringCharacters">string of characters</param>
+		/// <param name="i">index</param>
+		/// <returns>return true or false</returns>
+		private bool validOpenChars(string stringCharacters, int i) { 
+			return (stringCharacters[i] == '(' || stringCharacters[i] == '{' || stringCharacters[i] == '[') ? true : false;
+		}
+		/// <summary>
+		/// Validate start of string with closing characters and string is empty
+		/// </summary>
+		/// <param name="stringCharacters">string of characters</param>
+		/// <returns>return true or false</returns>
+		private bool validCloseOrEmptyChars(string stringCharacters) {
```

Comment:
there are a couple of things you could improve here:
the function name + parameter suggests it will compare characters (in plural) to close or empty chars.
however the function only validates the starting character (position 0).
If you see your 2 functions validOpenChars and validCloseChars, the name looks similar but they do different things.

So my suggestion would be change the function name to something like `isFirstCharCloseOrEmpty` or do something similar than with validOpenChars, including the position

Reply:
The other thing, is that this line is redundant:
return (boolean statement) ? true : false;

you can always `return boolean statement`